pkg/services/cloudmigration/cloudmigrationimpl/cloudmigration.go

```diff
@@ -464,14 +466,13 @@ func (s *Service) GetMigrationRunList(ctx context.Context, migUID string) (*clou
 	return runList, nil
 }
 
-func (s *Service) DeleteSession(ctx context.Context, uid string) (*cloudmigration.CloudMigrationSession, error) {
-	c, err := s.store.DeleteMigrationSessionByUID(ctx, uid)
+func (s *Service) DeleteSession(ctx context.Context, sessionUID string) (*cloudmigration.CloudMigrationSession, error) {
+	c, err := s.store.DeleteMigrationSessionWithRelatedElements(ctx, sessionUID)
```

Comment:
I think we should just add your functionality from your new function into the existing `DeleteMigrationSessionByUID` function, and just add documentation noting that the function deletes related elements.